include/af/opencl.h

```diff
@@ -147,6 +194,61 @@ namespace afcl
  }
 #endif
 
+#if AF_API_VERSION >= 33
+/**
+   Push user provided device control constructs into the ArrayFire device manager pool
+
+   This function should be used only when the user would like ArrayFire to use an
+   user generated OpenCL context and related objects for ArrayFire operations.
+
+   \param[in] dev is the OpenCL device for which user provided context will be used by ArrayFire
+   \param[in] ctx is the user provided OpenCL cl_context to be used by ArrayFire
+   \param[in] que is the user provided OpenCL cl_command_queue to be used by ArrayFire. If this
+                  parameter is NULL, then we create a command queue for the user using the OpenCL
+                  context they provided us.
+
+   \note The cl_* objects are passed onto c++ objects (cl::Device, cl::Context & cl::CommandQueue)
+   that are defined in the `cl.hpp` OpenCL c++ header provided by Khronos Group Inc. Therefore, please
+   be aware of the lifetime of the cl_* objects before passing them to ArrayFire.
+*/
+static inline void pushDevice(cl_device_id dev, cl_context ctx, cl_command_queue que)
+{
+    af_err err = afcl_push_device_context(dev, ctx, que);
+    if (err!=AF_SUCCESS) throw af::exception("Failed to push user provided device/context to ArrayFire pool");
+}
+
+/**
+   Set active device using cl_context and cl_device_id
+
+   \param[in] dev is the OpenCL device id that is to be set as Active device inside ArrayFire
+   \param[in] ctx is the OpenCL cl_context being used by ArrayFire
+*/
+static inline void setDevice(cl_device_id dev, cl_context ctx)
```

Comment:
Use #if AF_API_VERSION for each function separately. That's the style we are following.